### PR TITLE
Pipeline Aggregations at the root of the agg tree are now validated

### LINKED ISSUE #13179: Top-level pipeline aggs are not validated
If a pipeline is defined as a top-level agg (e.g a `sum_bucket` at the very root), it won't have any parent aggregations and thus sidesteps the current parser validation in AggregatorsParser.  The final clause which deals with top-level pipelines needs extra logic to validate pipelines ... but it will likely require some changes since the current pipeline validation methods require an agg parent.

/cc @colings86

### PR DESCRIPTION
**Pull request by colings86**
Previously PipelineAggregatorFactory's at the root to the agg tree (top-level aggs) were not validated. This commit adds a call to PipelineAggregatoFactory.validate() for that case.

Closes #13179

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java [hunk 1]
```diff
@@ -244,5 +244,13 @@ private void resolvePipelineAggregatorOrder(Map<String, AggregatorFactory> aggFa
                 orderedPipelineAggregators.add(factory);
             }
         }
+
+        AggregatorFactory[] getAggregatorFactories() {
+            return this.factories.toArray(new AggregatorFactory[this.factories.size()]);
+        }
+
+        List<PipelineAggregatorFactory> getPipelineAggregatorFactories() {
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java]: Feels a bit weird that one method is returning a list and the other one an array

Comment (colings86) [inline core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java]: agreed, should I return a list from both and do the conversion to an array in the AggParsers class?

Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java]: Why do we need to convert to an array? In general I think we should try to use either lists or arrays everywhere instead of doing back-and-forth conversions? (not for the cost of the conversions, more for consistency) Maybe this should be a different PR though, it looks like an existing issue to me. It was just weird when looking at the diff

**Comment by jpountz** [comment]
LGTM

File: core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java (hunk 1 above)
Comment (colings86) [inline core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java]: I agree and would prefer to use a list everywhere for aggs but I would rather not tackle that in this PR. I think the idea is that its a list here in the Builder as it needs to be dynamic so you can call addAggregator() but in the build() method it becomes an array because it doesn't change after that.
